hera_cal/apply_cal.py

```diff
@@ -2,17 +2,19 @@
 import hera_cal.io as io
 from pyuvdata import UVCal, UVData
 import argparse
+from hera_cal.datacontainer import DataContainer
 
 
-def recalibrate_in_place(data, data_flags, new_gains, cal_flags, old_gains=None, gain_convention = 'divide'):
+def recalibrate_in_place(data, data_flags, new_gains, cal_flags, old_gains=None, gain_convention='divide'):
     '''Update data and data_flags in place, taking out old calibration solutions, putting in
     new calibration solutions, and updating flags from those calibration solutions. Previously 
     flagged data is left unmodified. Missing antennas from either the new gains or (if it's not None),
     the old gains are automatically flagged in the data's visibilities that involves those antennas.
     
     Arguments:
         data: DataContainer containing baseline-pol complex visibility data. This is modified in place.
-        data_flags: DataContainer containing data flags. This is modified in place.
+        data_flags: DataContainer containing data flags. This is modified in place. Can also be fed as a
+            data weights dictionary with float dtype.
```

Comment:
If it is a float dtype, you should explain how the calibration operates (namely, that it treats weights == 0 as flags, but otherwise keeps weights the same).